src/modules/directory.rs

```diff
@@ -101,169 +121,863 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+        dir.truncate(3);
+        assert_eq!(dir.to_separator(Some("/"), None), "~/starship")
+    }
+
+    #[test]
+    fn truncate_same_path_as_provided_length() {
+        let mut dir = Directory::from("~/starship/engines");
+        dir.truncate(3);
+        assert_eq!(dir.to_separator(Some("/"), None), "~/starship/engines")
+    }
+
+    #[test]
+    fn truncate_slightly_larger_path_than_provided_length() {
+        let mut dir = Directory::from("~/starship/engines/booster");
+        dir.truncate(3);
+        assert_eq!(
+            dir.to_separator(Some("/"), None),
+            "starship/engines/booster"
+        )
+    }
+
+    #[test]
+    fn truncate_larger_path_than_provided_length() {
+        let mut dir = Directory::from("~/starship/engines/booster/rocket");
+        dir.truncate(3);
+        assert_eq!(dir.to_separator(Some("/"), None), "engines/booster/rocket")
+    }
+
+    #[test]
+    fn truncate_same_path_as_provided_length_from_root_unix() {
+        let mut dir = Directory::from("/starship/engines/booster");
+        dir.truncate(3);
+        assert_eq!(
+            dir.to_separator(Some("/"), None),
+            "/starship/engines/booster"
+        );
+    }
+
+    #[test]
+    fn truncate_same_path_as_provided_length_from_root_windows_forward_slash() {
+        let mut dir = Directory::from(r"C:/starship/engines/booster");
+        dir.truncate(3);
+        assert_eq!(
+            dir.to_separator(Some(r"\"), None),
+            r"C:\starship\engines\booster"
+        );
+    }
+
+    #[test]
+    #[cfg(target_os = "windows")]
+    fn truncate_same_path_as_provided_length_from_root_windows() {
+        let mut dir = Directory::from(r"C:\starship\engines\booster");
+        dir.truncate(3);
+        assert_eq!(
+            dir.to_separator(Some(r"\"), None),
+            r"C:\starship\engines\booster"
+        );
+    }
 
-        let output = contract_path(full_path, home, "~");
-        assert_eq!(output, "~/schematics/rocket");
+    #[test]
+    fn truncate_larger_path_than_provided_length_from_root_unix() {
+        let mut dir = Directory::from("/starship/engines/booster/rocket");
+        dir.truncate(3);
+        assert_eq!(dir.to_separator(Some("/"), None), "engines/booster/rocket");
+    }
+    #[test]
+    fn truncate_larger_path_than_provided_length_from_root_windows_forward_slash() {
+        let mut dir = Directory::from(r"C:/starship/engines/booster/rocket");
+        dir.truncate(3);
+        assert_eq!(
+            dir.to_separator(Some(r"\"), None),
+            r"engines\booster\rocket"
+        );
     }
 
     #[test]
     #[cfg(target_os = "windows")]
+    fn truncate_larger_path_than_provided_length_from_root_windows() {
+        let mut dir = Directory::from(r"C:\starship\engines\booster\rocket");
+        dir.truncate(3);
+        assert_eq!(
+            dir.to_separator(Some(r"\"), None),
+            r"engines\booster\rocket"
+        );
+    }
+
+    #[test]
```

Comment:
Shouldn't this be windows-only as well?

Reply:
I am afraid there has been too much changes to this module's code since this PR was submitted.
Not sure it can still be merged now after so much time in the backlog.